docs/user/testing.md

```diff
@@ -0,0 +1,182 @@
+# Testing
+
+In this section, we will talk about how to test different modules and algorithms in garage.
+
+In garage, we use [pytest](https://docs.pytest.org/en/stable/getting-started.html) to carry out tests.
+
+## Test modules
+
+All test files are stored under `tests/` in the garage respository.
+
+Test modules are organized in the same way as the garage main repository. Ideally, all main modules and files under `src/garage/` should be covered with test cases.
+
+Test modules of garage are structured in the following ways:
+
+- `tests/` contains all tests and supporting code
+- `tests/garage` contains unit tests
+- `tests/integration_tests` contains integration tests
+- modules in `tests/` and `tests/fixtures` contain helper codes and fixtures which make writing good tests easier.
+
+```bash
+├── tests
+│   ├── fixtures
+│   ├── garage
+│   └── integration_tests
+├── helpers.py
+└── (helper_files.py)
+```
+
+## Prerequisites
```

Comment:
i think you should mention that none of this will work unless you install `garage[dev]`

Reply:
Right! I think I'll also include  `pip install 'garage[all,dev]'` since there are examples and tests which require mujoco and dm_control package.